app/services/oidc_jwt_token_validate.rb

```diff
@@ -32,7 +32,8 @@ def generate_jwks_uri
       :verify_ssl => true
       )
     rsa_public = JSON.parse(response)['keys'].first
-    generate_key_from_jwks(rsa_public)
+    jwk = JSON::JWK.new(rsa_public)
```

Comment:
IIUC, this is supported by jwt gem also so we don't need the extra dependency - https://github.com/jwt/ruby-jwt#json-web-key-jwk?

Reply:
@tbrisker  this feature `JWT::JWK` was broken for the `JWT` gem, right now we are using the 2.1.0 version, and this broken feature has been worked upon and fixed in their 2.2.0 version. Right now the gem is in its 2.2.0 beta version. 

So what do you suggest? we should wait for them or use a new gem as done here?

Reply:
Good news! if there are no issues/blockers reported, then the next release for the JWT gem with the `JWT::JWK` fix will be available next week :+1:

Reply:
Last I checked JWT::JWK requires a public key as input, but keycloak provides a json rather than the key itself. Therefore, here the use of JSON::JWK would be required.

Reply:
aren't you already pulling the key out of the json one line above?

Reply:
please rename the variable so it's clear that it's a json object and not the actual key, i was confused by the naming.